collect_app/src/main/java/org/odk/collect/android/tasks/FormLoaderTask.java

```diff
@@ -182,25 +184,17 @@ protected FECWrapper doInBackground(String... path) {
         }
 
         // Remove previous forms
-        ReferenceManager.instance().clearSession();
+        referenceManager.clearSession();
 
         processItemSets(formMediaDir);
 
         // This should get moved to the Application Class
-        if (ReferenceManager.instance().getFactories().length == 0) {
+        if (referenceManager.getFactories().length == 0) {
             // this is /sdcard/odk
-            ReferenceManager.instance().addReferenceFactory(new FileReferenceFactory(Collect.ODK_ROOT));
+            referenceManager.addReferenceFactory(new FileReferenceFactory(Collect.ODK_ROOT));
         }
 
-        // Set jr://... to point to /sdcard/odk/forms/filename-media/
-        ReferenceManager.instance().addSessionRootTranslator(
-                new RootTranslator("jr://images/", "jr://file/forms/" + formFileName + "-media/"));
-        ReferenceManager.instance().addSessionRootTranslator(
-                new RootTranslator("jr://image/", "jr://file/forms/" + formFileName + "-media/"));
-        ReferenceManager.instance().addSessionRootTranslator(
-                new RootTranslator("jr://audio/", "jr://file/forms/" + formFileName + "-media/"));
-        ReferenceManager.instance().addSessionRootTranslator(
-                new RootTranslator("jr://video/", "jr://file/forms/" + formFileName + "-media/"));
+        addSessionRootTranslators(formFileName, referenceManager, "images", "image", "audio", "video");
```

Comment:
Why do you call this method two times? Can't it be 
`addSessionRootTranslators(formFileName, referenceManager, "images", "image", "audio", "video", "file");
`
in line 131 and then we wouldn't need it here at all?

Reply:
I'm guessing that it's because the session gets cleared in line 187, but I believe we could clean the session before line 131 and then do what @grzesiek2010 suggests. It doesn't hurt to have a translator for `jr://file` and the other hosts (images, image, audio, video) at the same time.

Reply:
We would also need the `if` block at 192 moved before line 131 too, right?

Reply:
That's right.

Reply:
>Why do you call this method two times?

That’s temporary. I don’t yet understand the impact of adding them all earlier.